lib/scrapers/redditscrape.py

```diff
@@ -0,0 +1,105 @@
+import praw
+from tornado import gen
+from lib.config import CONFIG
+
+
+class RedditScraper(object):
+    name = 'reddit'
+
+    @property
+    def num_scrape(self):
+        if CONFIG['_mode'] == 'prod':
+            return 1000
+        return 100
+
+    @gen.coroutine
+    def scrape(self, user_data):
+        try:
+            tokens = user_data.services['reddit']
+        except KeyError:
+            return False
+        if 'denied' in tokens:
+            return False
+        client_id = CONFIG.get("reddit_client_id"),
+        client_secret = CONFIG.get("reddit_client_secret")
+
+        r = praw.Reddit(user_agent="QS-agent/1.0")
+        r.set_oauth_app_info(client_id=client_id, client_secret=client_secret, redirect_uri="https://iamadatapoint.com/auth/reddit")
+        r.set_access_credentials(**tokens)
```

Comment:
No not on gulper which is why I'm saying tokens may be incorrectly
formatted or named wrong or something
On Apr 1, 2016 20:51, "Micha Gorelick" <notifications@github.com> wrote:

> In lib/scrapers/redditscrape.py
> <https://github.com/QuantifiedSelfless/gulper/pull/5#discussion_r58287269>
> :
>
> > +        return 100
> > +
> > +    @gen.coroutine
> > +    def scrape(self, user_data):
> > +        try:
> > +            tokens = user_data.services['reddit']
> > +        except KeyError:
> > +            return False
> > +        if 'denied' in tokens:
> > +            return False
> > +        client_id = CONFIG.get("reddit_client_id"),
> > +        client_secret = CONFIG.get("reddit_client_secret")
> > +
> > +        r = praw.Reddit(user_agent="QS-agent/1.0")
> > +        r.set_oauth_app_info(client_id=client_id, client_secret=client_secret, redirect_uri="https://iamadatapoint.com/auth/reddit")
> > +        r.set_access_credentials(**tokens)
>
> you sure? did you test that on gulper?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/QuantifiedSelfless/gulper/pull/5/files/f1450f9648e44cffba41d84cf5e5f3db98937a88#r58287269>
>